File: 로또의_최고_순위와_최저_순위.py
def solution(lottos, win_nums):
    count = 0
    zero = 0
    for n in lottos:
        if n == 0:
            zero+=1
        else:
            if n in win_nums:
                count+=1
    highRank = 7-count-zero
    lowRank = 7-count
    if highRank>6:
        highRank=6
    if lowRank>6:
        lowRank=6
    return [highRank, lowRank]

File: java/Level 1/로또의_최고_순위와_최저_순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        
        int zero = 0;
        int match = 0;
        
        for(int i: lottos){
            if(i == 0){
                zero += 1;
            }
            else{
                for(int n: win_nums){
                    if(i == n){
                        match += 1;
                    }
                }
            }
        }
        
        int max_rank = Math.min(7 - zero - match, 6);
        int min_rank = Math.min(7 - match, 6);
        
        answer[0] = max_rank;
        answer[1] = min_rank;
        return answer;
    }
}
